internal/logger/telemetry_store.go

```diff
@@ -72,6 +85,18 @@ type TelemetryStore struct {
 
 	// Rules evaluated during processing
 	Evals []RuleEvalData `json:"rules"`
+
+	// Repository is the repository ID that the request was associated with.
+	Repository *uuid.UUID `json:"repository"`
```

Comment:
Do we benefit from storing these as `uuid.UUID`s rather than `string`s?  I see two related problems:

1. It means that we need to call `uuid.Parse` to extract this data from the original protobufs where it makes sense.
2. It adds additional error cases when recording information, because it's possible that some other part of the system interprets UUIDs differently than this code.  (Unlikely in this case, but if we decided to change the format, this is one more place we'd need to update.)

If the data is already generally stored in the code as a uuid.UUID, then this seems fine, but if we're needing to add a bunch of parsing calls, I'd tend to prefer to store this as a `string` and make the logging code as dumb as possible, and then do cleanup elsewhere -- that makes it possible for us to see cases where the cleanup didn't happen correctly when we're observing corrupted data.

Reply:
BTW I'll rebase my changes on top of what Ria made, so I'm available to incorporate more feedback on this before merging 😃

Reply:
It's useful if we're mostly storing something that's already a `uuid.UUID` into the record, like we do for Project in some cases.  In other cases (e.g. line 130 of `handlers_artifacts.go`), we end up parsing the UUID particularly for the telemetry recording, introducing a new error case that we previously never cared about.

It looks like we only do that once, though, so I think I over-reacted based on a sample size of 2-3 at the start of the review.